File: java/selenium/src/test/java/com/github/bmsantos/selenium/BaseColaTest.java
package com.github.bmsantos.selenium;

import com.github.bmsantos.core.cola.story.annotations.IdeEnabler;
import org.junit.AfterClass;
import org.junit.Test;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;

import static java.util.concurrent.TimeUnit.SECONDS;
import static org.assertj.core.api.Assertions.fail;

public abstract class BaseColaTest {

    public static String baseUrl = "http://google.com";
    private static WebDriver driver;

    public BaseColaTest() {
        if (driver == null) {
            createDriver();
        }
    }

    @AfterClass
    public static void quitDriver() {
        if (driver != null) {
            driver.quit();
            driver = null;
        }
    }

    public WebDriver getDriver() {
        return driver;
    }

    private void createDriver() {
        driver = new FirefoxDriver();
        driver.manage().timeouts().implicitlyWait(10, SECONDS);
    }

    @IdeEnabler
    @Test
    public void thisTestWillNotExecute() {
        fail("This test failed because it is not supposed to be executed!");
    }
}

File: kotlin/kotlin-cola-tests/src/test/java/cola/ide/BaseColaTest.kt
package cola.ide

import com.github.bmsantos.core.cola.story.annotations.IdeEnabler
import org.junit.Test

/**
 * Created with IntelliJ IDEA.
 * Date: 12/2/14
 * Time: 5:15 PM
 */
open class BaseColaTest() {
    @IdeEnabler
    @Test
    public fun iWillBeRemoved() {
        println("This test should have not been executed")
    }
}